portal/models/communication.py

```diff
@@ -244,14 +229,49 @@ def generate_and_send(self):
                 "{} contains unknown varables: {}".format(
                     self.communication_request.content_url,
                     ','.join(missing)))
-        self.message = EmailMessage(
+
+        msg = EmailMessage(
             subject=mailresource.subject,
             body=mailresource.body,
             recipients=user.email,
+            sender=current_app.config['MAIL_DEFAULT_SENDER'],
             user_id=user.id)
+
+        return msg
+
+
+    def generate_and_send(self):
+        "Collate message details and send"
+
+        if current_app.config.get('DEBUG_EMAIL', False):
+            # hack to restart trace when in loop from celery task
+            # don't want to reset if in the middle of a request
+            from celery import current_task
+            establish_trace(
+                "BEGIN trace as per DEBUG_EMAIL configuration",
+                reset_trace=current_task is None)
+
+        user = User.query.get(self.user_id)
+        if not user.email or '@' not in user.email:
```

Comment:
Does the `@` need to be unicode (`u'@'`)?

Reply:
It looks like this doesn't matter either way (`'@' in 't@st'`, `u'@' in 't@st'`, `'@' in u't@st'`, and `u'@' in u't@st'` all return `True`), but I'll change it anyway